dbms/src/TiDB/Schema/tests/gtest_schema_sync.cpp

```diff
@@ -0,0 +1,326 @@
+#include <Debug/MockTiDB.h>
+#include <Interpreters/InterpreterDropQuery.h>
+#include <Parsers/ASTDropQuery.h>
+#include <Parsers/IAST.h>
+#include <Storages/ColumnsDescription.h>
+#include <Storages/IManageableStorage.h>
+#include <Storages/Transaction/RegionBlockReader.h>
+#include <Storages/Transaction/TMTContext.h>
+#include <Storages/Transaction/Types.h>
+#include <Storages/registerStorages.h>
+#include <TestUtils/TiFlashTestBasic.h>
+#include <TestUtils/TiFlashTestEnv.h>
+#include <TiDB/Schema/SchemaSyncService.h>
+#include <common/defines.h>
+
+namespace DB
+{
+namespace FailPoints
+{
+extern const char exception_before_rename_table_old_meta_removed[];
+extern const char force_context_path[];
+} // namespace FailPoints
+namespace tests
+{
+
+class SchemaSyncTest : public ::testing::Test
+{
+public:
+    SchemaSyncTest()
+        : global_ctx(TiFlashTestEnv::getGlobalContext())
+    {}
+
+    static void SetUpTestCase()
+    {
+        try
+        {
+            registerStorages();
+        }
+        catch (DB::Exception &)
+        {
+            // Maybe another test has already registed, ignore exception here.
+        }
+
+        FailPointHelper::enableFailPoint(FailPoints::force_context_path);
+    }
+
+    static void TearDownTestCase()
+    {
+        FailPointHelper::disableFailPoint(FailPoints::force_context_path);
+    }
+
+    void SetUp() override
+    {
+        // disable schema sync timer
+        global_ctx.getSchemaSyncService().reset();
+        recreateMetadataPath();
+    }
+
+    void TearDown() override
+    {
+        for (auto & [db_name, db_id] : MockTiDB::instance().getDatabases())
+        {
+            UNUSED(db_id);
+            MockTiDB::instance().dropDB(global_ctx, db_name, false);
+        }
+        for (auto & db : global_ctx.getDatabases())
+        {
+            mustDropSyncedDatabase(db.first);
+        }
+
+        // restore schema sync timer
+        if (!global_ctx.getSchemaSyncService())
+            global_ctx.initializeSchemaSyncService();
+    }
+
+    void refreshSchema()
+    {
+        auto & flash_ctx = global_ctx.getTMTContext();
+        auto schema_syncer = flash_ctx.getSchemaSyncer();
+        try
+        {
+            schema_syncer->syncSchemas(global_ctx);
+        }
+        catch (Exception & e)
+        {
+            if (e.code() == ErrorCodes::FAIL_POINT_ERROR)
+            {
+                return;
+            }
+            else
+            {
+                throw;
+            }
+        }
+    }
+
+    void resetSchemas()
+    {
+        auto & flash_ctx = global_ctx.getTMTContext();
+        flash_ctx.getSchemaSyncer()->reset();
+    }
+
+    ManageableStoragePtr mustGetSyncedTable(TableID table_id)
+    {
+        auto & flash_ctx = global_ctx.getTMTContext();
+        auto & flash_storages = flash_ctx.getStorages();
+        auto tbl = flash_storages.get(table_id);
+        RUNTIME_CHECK(tbl, "Can not find table in TiFlash instance!", table_id);
```

Comment:
```suggestion
        RUNTIME_CHECK_MSG(tbl, "Can not find table {} in TiFlash instance!", table_id);
```

Reply:
I get it. The error message is duplicated if we use RUNTIME_CHECK
`DB::Exception: Check tbl failed, "Can not find table in TiFlash instance!" = Can not find table in TiFlash instance!, table_id = 36`